src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -580,6 +600,58 @@ def _deploy_arm_template_at_tenant_scope(cli_ctx,
                        deployment_name, deployment_properties, deployment_location)
 
 
+def what_if_deploy_arm_template_at_resource_group(cmd, resource_group_name,
+                                                  template_file=None, template_uri=None, parameters=None,
+                                                  deployment_name=None, mode=DeploymentMode.incremental,
+                                                  aux_tenants=None, result_format=None,
+                                                  no_pretty_print=None, no_prompt=False):
+    what_if_properties = _prepare_deployment_what_if_properties(cmd.cli_ctx, template_file, template_uri,
+                                                                parameters, mode, result_format, no_prompt)
+    mgmt_client = _get_deployment_management_client(cmd.cli_ctx, aux_tenants=aux_tenants)
+    what_if_poller = mgmt_client.what_if(resource_group_name, deployment_name, what_if_properties)
+
+    return _what_if_deploy_arm_template_core(cmd.cli_ctx, what_if_poller, no_pretty_print)
+
+
+def what_if_deploy_arm_template_at_subscription_scope(cmd,
+                                                      template_file=None, template_uri=None, parameters=None,
+                                                      deployment_name=None, deployment_location=None,
+                                                      result_format=None, no_pretty_print=None, no_prompt=False):
+    what_if_properties = _prepare_deployment_what_if_properties(cmd.cli_ctx, template_file, template_uri, parameters,
+                                                                DeploymentMode.incremental, result_format, no_prompt)
+    mgmt_client = _get_deployment_management_client(cmd.cli_ctx)
+    what_if_poller = mgmt_client.what_if_at_subscription_scope(deployment_name, what_if_properties, deployment_location)
+
+    return _what_if_deploy_arm_template_core(cmd.cli_ctx, what_if_poller, no_pretty_print)
+
+
+def _what_if_deploy_arm_template_core(cli_ctx, what_if_poller, no_pretty_print):
+    what_if_result = LongRunningOperation(cli_ctx)(what_if_poller)
+
+    if no_pretty_print:
+        return what_if_result
+
+    try:
+        if cli_ctx.enable_color:
+            # Diabling colorama since it will silently strip out the Xterm 256 color codes the What-If formatter
+            # is using. Unfortuanately, the colors that colorama supports are very limited, which doesn't meet our needs.
+            from colorama import deinit
+            deinit()
+
+            # Enable virtual terminal mode for Windows console so it processes color codes.
+            if platform.system() == "Windows":
+                from ._win_vt import enable_vt_mode
+                enable_vt_mode()
+
+        print(format_what_if_operation_result(what_if_result, cli_ctx.enable_color))
+    finally:
+        if cli_ctx.enable_color:
+            from colorama import init
+            init()
```

Comment:
Colorama usage part looks good to me.

Reply:
Thanks for reviewing it, Jiashuo!

Reply:
BTW, here is the discussion on Colorama: https://github.com/tartley/colorama/issues/160#issuecomment-372687727

Reply:
That would be awesome 😀! Btw I think I found out an small bug related to colorama while test it the pipe with CLI. I'll open an issue in the knack repo soon.